File: src/main/java/com/example/examplemod/codegen/ExampleModItemModelProvider.java
package com.example.examplemod.codegen;

import com.example.examplemod.item.ExampleItems;
import net.minecraft.data.DataGenerator;
import net.minecraft.item.Item;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.client.model.generators.ItemModelProvider;
import net.minecraftforge.common.data.ExistingFileHelper;

import java.util.Objects;

public class ExampleModItemModelProvider extends ItemModelProvider {
    public ExampleModItemModelProvider(DataGenerator gen, String modId, ExistingFileHelper existingFileHelper) {
        super(gen, modId, existingFileHelper);
    }

    @Override
    protected void registerModels() {
        simpleItem(ExampleItems.TITANIUM_INGOT.get());
    }

    private void simpleItem(Item item) {
        ResourceLocation name = Objects.requireNonNull(item.getRegistryName());
        singleTexture(name.getPath(), mcLoc(folder + "/generated"), "layer0", new ResourceLocation(name.getNamespace(), folder + "/" + name.getPath()));
    }
}

File: src/main/java/com/example/examplemod/ExampleModItemModelProvider.kt
package com.example.examplemod

import net.minecraft.data.DataGenerator
import net.minecraft.item.Item
import net.minecraftforge.client.model.generators.ItemModelProvider
import net.minecraftforge.common.data.ExistingFileHelper
import net.minecraft.util.ResourceLocation
import java.util.*


class ExampleModItemModelProvider(gen: DataGenerator, modId: String, existingFileHelper: ExistingFileHelper?) :
    ItemModelProvider (gen,modId,existingFileHelper){
    override fun registerModels() {
        simpleItem(ExampleItems.TITANIUM_INGOT.get())
    }

    private fun simpleItem(item: Item) {
        val name: ResourceLocation = item.registryName!!
        singleTexture(
            name.path,
            mcLoc("$folder/generated"),
            "layer0",
            ResourceLocation(name.namespace, folder + "/" + name.path)
        )
    }
}
